### PR TITLE
General Silicon Fixes

### LINKED ISSUE #7151: ais cant use shells anymore after dying in one
Issue reported from  Round ID: 26507 (Yogstation 13)

Reporting client version: 512.1488

image
What the title says. AI shell dies to darkspawn with a flash and a ranged stun and a lot of melee bashing, CE repairs and reboots it. I try to enter it using all three of my verbs and also clicking on it, all return the same message.
Another shell is made. The other AIs don't claim it. I cannot enter it either. I notice my core has a "connected to shell" icon on its diagnostic HUD.
image

### PR DESCRIPTION
**Pull request by redmoogle**
### Intent of your Pull Request

1. Fixes: #7151
2. Fixes irrelevant runtime where byond is too dumb to return null for a empty list
3. Cleans the code up a little
4. Removes a duplicated piece of code _Bibby_

### Why is this good for the game?

Fixes are good, yeh? 

#### Changelog

:cl:  
bugfix: Dying as an AI shell nolonger bricks that shell
/:cl:

### CODE REVIEW

**Comment by redmoogle** [comment]
Tested

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -586,36 +586,31 @@
...
 	//check if it doesn't require any access at all
 	if(check_access(null))
-		return 1
+		return TRUE
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
 		//if they are holding or wearing a card that has access, that works
 		if(check_access(H.get_active_held_item()) || check_access(H.wear_id))
-			return 1
-	else if(ismonkey(M))
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/robot/robot.dm:595]: keep :/ 

no removey codey!!

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 2]
```diff
@@ -1249,6 +1246,3 @@
 		old_ai.connected_robots -= src
 	if(connected_ai)
 		connected_ai.connected_robots |= src
-
-/mob/living/silicon/robot/shell
```
Comment (TheGamerdk) [inline code/modules/mob/living/silicon/robot/robot.dm:1253]: ?

**Review by jcatjmeow — APPROVED** [review]
Looks fine, ran without issues

**Pull request merged**